internal/ingress/annotations/annotations.go

```diff
@@ -205,3 +206,12 @@ func ExtractMethods(anns map[string]string) []string {
 	}
 	return strings.Split(val, ",")
 }
+
+// ExtractSNI extracts the route SNI match criteria annotation value.
+func ExtractSNIs(anns map[string]string) ([]string, bool) {
+	val, exists := anns[AnnotationPrefix+SNIKey]
+	if val == "" {
+		return []string{}, exists
```

Comment:
```suggestion
		return nil, exists
```

`nil` is a valid slice, equivalent to an empty slice, that doesn't take up memory. It's idiomatic to use `nil` instead of empty slices.